src/lib.rs

```diff
@@ -54,11 +63,20 @@ mod helpers;
 mod html_root_url;
 mod markdown_deps;
 
+// Hacky workaround to ensure that at least one feature is enabled
+#[cfg(not(any(feature = "regex_version", feature = "html_root_url",
+    feature = "markdown")))]
+const AT_LEAST_ONE_FEATURE_ENABLED: u32 = "bad";
```

Comment:
I don't think anything strange actually happens if you disable all features? You just get what you ask for, i.e., no macros and thus no functionality.

The real way to do this seems to be to use [`compile_error`](https://doc.rust-lang.org/std/macro.compile_error.html).

Reply:
I changed this to use `compile_error`. The reason I added this in is to alert anyone who uses `default-features = false` by default that they would need to enable at least one feature in order to use the corresponding macro(s) this crate provides.

(I'm leaving this unresolved in case you want to discuss this further.)

Reply:
I have been thinking that it would be nice if the library had some kind of functionality even without the features. That functionality could be a `assert_contains_version!` macro which simply does a normal string match to search for a version number.

That would make the features nicely additive and it would also ensure that the functions in `helpers.rs` have a purpose even when all features are disabled.

We don't need to do this in this PR, but I think it could be a good preparation which would simplify this PR a tiny bit (less need for `[cfg(...)]` for most of the function which use `std` functions).

Reply:
My opinion is that the `compile_error` and other remaining `#[cfg(...)]` blocks on imports can be removed when that basic functionality gets added, but I'm open to arguments otherwise.

Reply:
After a bit more thought I removed the remaining `use` statements by fully qualifying the function declarations that used them.

Reply:
Looks nice now, thanks!